p2p/host/autorelay/relay_finder.go

```diff
@@ -145,18 +146,29 @@ func (rf *relayFinder) background(ctx context.Context) {
 				return
 			}
 			evt := ev.(event.EvtPeerConnectednessChanged)
-			if evt.Connectedness != network.NotConnected {
+
+			if evt.Connectedness == network.NotConnected {
+				rf.relayMx.Lock()
+				if rf.usingRelay(evt.Peer) { // we were disconnected from a relay
+					log.Debugw("disconnected from relay", "id", evt.Peer)
+					delete(rf.relays, evt.Peer)
+					rf.notifyMaybeConnectToRelay()
+					rf.notifyMaybeNeedNewCandidates()
+					push = true
+				}
+				rf.relayMx.Unlock()
+			} else if evt.Connectedness == network.Connected {
+				for _, r := range rf.conf.staticRelays {
+					if r.ID == evt.Peer {
+						rf.notifyMaybeConnectToRelay()
```

Comment:
I don't follow why we need this logic now. Aren't we already connected?

Reply:
That was the attempt to re-reserve slots when we are back online (when the client had intermitent network connection). In that case, the relay [would have dropped all our reservations](https://github.com/libp2p/go-libp2p/blob/master/p2p/protocol/circuitv2/relay/relay.go#L581-L591) and we need to re-reserve